src/util/normalize-manifest/resolve-relative.js

```diff
@@ -0,0 +1,52 @@
+import {FILE_PROTOCOL_PREFIX} from '../../resolvers/exotics/file-resolver.js';
+import {LINK_PROTOCOL_PREFIX} from '../../resolvers/exotics/link-resolver.js';
+
+const path = require('path');
+
+export default function(info: Object, moduleLoc: string, lockfileFolder: string): void {
+  // It won't work if we don't yet know what's the folder we'll use as root. It's not a
+  // big deal tho, because it only happens when trying to figure out the root, and we
+  // don't need to know the dependencies / devDependencies at this time.
+  if (!lockfileFolder) {
+    return;
+  }
+
+  for (const hint of ['dependencies', 'devDependencies', 'optionalDependencies', 'peerDependencies']) {
+    if (!info[hint]) {
+      continue;
+    }
+
+    for (const name of Object.keys(info[hint])) {
+      let value = info[hint][name];
+
+      if (path.isAbsolute(value)) {
+        value = FILE_PROTOCOL_PREFIX + value;
+      }
+
+      let prefix;
+
+      if (value.indexOf(FILE_PROTOCOL_PREFIX) === 0) {
+        prefix = FILE_PROTOCOL_PREFIX;
+      } else if (value.indexOf(LINK_PROTOCOL_PREFIX) === 0) {
+        prefix = LINK_PROTOCOL_PREFIX;
+      }
+
+      if (prefix) {
```

Comment:
Taking a level of indent, making the code closer to the left are both facts. I'd argue text closer to left side is easier to read text indented otherwise we'd have large margins on the left side for all text. That said I'm gonna stop here :D